src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnection.cs

```diff
@@ -449,6 +459,16 @@ private Task WriteFormattedInt32Async(int value, CancellationToken cancellationT
                     responseStream = EmptyReadStream.Instance;
                     ReturnConnectionToPool();
                 }
+                else if (request.Method == s_httpConnectMethod && (int)response.StatusCode == 200)
+                {
+                    // Successful response to CONNECT does not have body.
+                    // What ever comes next should be opaque.
+                    responseStream = new RawConnectionStream(this);
+                    // Put connection back to the pool if we upgraded to tunnel.
+                    // We cannot use it for normal HTTP requests any more.
+                    _connectionClose = true;
```

Comment:
Is this necessary?  I don't think we're going to put it back in the pool regardless.

Reply:
I can add test for it. Once you submit your updated loopback changes, it may be easier to test. 
I may miss something but I did not see any evidence that we would threat connection special after TLS upgrade.